zipline/data/us_equity_minutes.py

```diff
@@ -11,27 +11,402 @@
+    2016-01-20 21:00
+
+    All assets are written with a common 'index', sharing a common first
+    trading day. Assets that do not begin trading until after the first trading
+    day will have zeros for all pricing data up and until data is traded.
+
+    'index' is in quotations, because bcolz does not provide an index. The
+    format allows index-like behavior by writing each minute's data into the
+    corresponding position of the enumeration of the aforementioned datetime
+    index.
+
+    The datetimes which correspond to each position are written in the metadata
+    as integer nanoseconds since the epoch into the `minute_index` key.
+    """
+    def __init__(self,
+                 first_trading_day,
+                 rootdir,
+                 market_opens,
+                 minutes_per_day=MINUTES_PER_DAY,
+                 ohlc_ratio=OHLC_RATIO,
+                 expectedlen=DEFAULT_EXPECTEDLEN):
+        """
+        Parameters:
+        -----------
+
+        first_trading_day : datetime-like
+            The first trading day in the data set.
+        
+        rootdir : string
+            Path to the root directory into which to write the metadata and
+            bcolz subdirectories.
+
+        market_opens : pd.Series
+            The market opens used as a starting point for each periodic span of
+            minutes in the index.
+
+        minutes_per_day : int
+            The number of minutes per each period. Defaults to 390, the mode
+            of minutes in NYSE trading days.
+
+        ohlc_ratio : int
+            The ratio by which to multiply the pricing data to convert the floats
+            from floats to an integer to fit within the np.uint32.
+
+            The default is 1000 to support pricing data which comes in to the
+            thousands place.
+
+        expectedlen : int
+            The expected length of the dataset, used when creating the initial
+            bcolz ctable.
+
+            If the expectedlen is not used, the chunksize and corresponding
+            compression ratios are not ideal.
+
+            Defaults to supporting 15 years of market data.
+
+            see: http://bcolz.blosc.org/opt-tips.html#informing-about-the-length-of-your-carrays
+        """
+        self._rootdir = rootdir
+        self._first_trading_day = first_trading_day
+        self._market_opens = market_opens[
+            market_opens.index.slice_indexer(start=self._first_trading_day)]
+        self._trading_days = market_opens.index
+        self._minutes_per_day = minutes_per_day
+        self._expectedlen = expectedlen
+        self._ohlc_ratio = ohlc_ratio
+
+        self._minute_index = _calc_minute_index(
+            self._market_opens, self._minutes_per_day)
+
+        self._write_metadata()
+
+    @property
+    def first_trading_day(self):
+        return self._first_trading_day
+
+    @property
+    def metadata_path(self):
+        return os.path.join(self._rootdir, METADATA_FILENAME)
```

Comment:
Seems like this path construction logic should be broken out into a function that's shared between the reader and the writer, ala:

```
def bcolz_metadata_path(rootdir):
    return os.path.join(rootdir, METADATA_FILENAME)
```